### ISSUE TITLE
File naming for python

### ISSUE THREAD

**Issue by juniwang** [Azure/autorest#2874]
Our management client and models start with `SignalR`. The generated files are named `signal_rXXX` (e.g. `signal_roperations.py`, `signal_rmanagement_client.py`, `signal_rresource.py`)which are very weird.  Is it possible that we configure autorest to generate files that are named `signalr_XXX`? We think `signalr_resource.py` makes much more sense than `signal_rresource.py`.

**Comment by lmazuel** [comment]
That's the expected behavior, based on the capital case you use. If you want custom naming, you can use `x-ms-client-name`:
https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-client-name

**Comment by fearthecowboy** [comment]
I'm not sure `x-ms-client-name` will work so well here -- the source of the `signal_r` in this case is the  `operationId`.

I think you're going to have to change the `operationId` fields to stuff like `"Signalr_ListOperations"`

**Comment by juniwang** [comment]
Thanks guys. I tried both but not perfectly. `x-ms-client-name` works on properties/parameters but not the file names. Acutally the class names or parameter names are quite good without extra extension.  The `operationId` partially work only for file `signalr_operations.py`. The files for generated models are still `signa_rXXX.py`.
Anyway, thanks. I tried to import the SDK modules from user's pespective, they actually aren't aware of the file names.  Well defined class names are enough. So we will leave the file names as `signal_rXXX`.
